src/HCSR04.cpp

```diff
@@ -20,21 +22,30 @@ double UltraSonicDistanceSensor::measureDistanceCm() {
 }
 
 double UltraSonicDistanceSensor::measureDistanceCm(float temperature) {
+    uint32_t maxDistanceDurationMicroSec;
+
     // Make sure that trigger pin is LOW.
     digitalWrite(triggerPin, LOW);
     delayMicroseconds(2);
     // Hold trigger for 10 microseconds, which is signal for sensor to measure distance.
     digitalWrite(triggerPin, HIGH);
     delayMicroseconds(10);
     digitalWrite(triggerPin, LOW);
+    double speedOfSoundInCmPerMicroSec = 0.03313 + 0.0000606 * temperature; // Cair ≈ (331.3 + 0.606 ⋅ ϑ) m/s
+
+    // Compute max delay based on max distance with 25% margin in microseconds or use absolute timeout
```

Comment:
"or use absolute timeout" -> that is not true any more, not we do minimum of those two. Maybe it is best to omit this last part, it is visible from the code.